CRM/Contribute/Page/ContributionPage.php

```diff
@@ -384,9 +384,18 @@ public function copy() {
       $this, TRUE, 0, 'GET'
     );
 
-    CRM_Contribute_BAO_ContributionPage::copy($gid);
+    $copy = CRM_Contribute_BAO_ContributionPage::copy($gid);
 
-    CRM_Utils_System::redirect(CRM_Utils_System::url(CRM_Utils_System::currentPath(), 'reset=1'));
+    $urlString = CRM_Utils_System::currentPath();
+    $urlParams = 'reset=1';
+
+    // Redirect to copied contribution page
+    if ($copy->id) {
+      $urlString = 'civicrm/admin/contribute/settings';
+      $urlParams .= '&action=update&id=' . $copy->id;
+    }
+
+    CRM_Utils_System::redirect(CRM_Utils_System::url($urlString, $urlParams));
```

Comment:
It would be better to have the calling function do this redirect stuff & have a BAO::copy function do the copy

Reply:
The BAO copy is doing the copying. This is just like a 'middleman' between the action being called, and copy actually happening. This is consistent with how events are being copied too. I can dig a bit deeper on refactoring if you like, to try and remove these 'middleman' copy methods.

Reply:
@eileenmcnaughton What @JKingsnorth has done here looks correct to me.  All the actual copy stuff is in the BAO copy object.  The redirect is specific to `CRM_Contribute_Page_ContributionPage` as it would be a different redirect if (eg.) called via an alternative form and/or API for managing contribution pages.

Reply:
@mattwire ok - sounds like you have it figured - can you merge if you are happy?